plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java

```diff
@@ -60,17 +60,22 @@
         }
 
         SchemaTableName sourceTableName = PROPERTIES.getSourceTableName(tableName);
-        Optional<Table> table = metadata.getMetastore().getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName());
-        if (table.isEmpty()) {
-            throw new TableNotFoundException(tableName);
-        }
-        if (isDeltaLakeTable(table.get())) {
+        return metadata.getMetastore()
+                .getTable(new HiveIdentity(session), sourceTableName.getSchemaName(), sourceTableName.getTableName())
+                .map(table -> getSystemTable(sourceTableName, table))
+                .orElseThrow(() -> new TableNotFoundException(tableName));
```

Comment:
The right pattern to do this would be

```
Table table = metadata.getMetastore().getTable(...)
   .orElseThrow(() -> new TableNotFoundException(tableName));
```

Reply:
You are right.
I addressed this in the meantime in https://github.com/trinodb/trino/pull/10447